packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -954,6 +958,47 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         return dataset.getData(...args);
     }
 
+    /**
+     * Retrieves all the data from the default crawler {@apilink Dataset} and exports them to the specified format.
+     * Supported formats are currently 'json' and 'csv', and will be inferred from the `path` automatically.
+     */
+    async exportData<Data>(path: string, format?: 'json' | 'csv'): Promise<Data[]> {
```

Comment:
I'm not sure I understand the difference between this and `exportTo`. My best guess is that this one saves to an arbitrary path, while `exportTo` uses the KV store, unless you say it should return an object. Is that correct? As a newcomer, I may not know which one to call.

Reply:
This is a helper on the crawler class, while `exportTo` is a `Dataset` method. 

> unless you say it should return an object

That if branch for `object` is just an implementation detail, as I needed only the data (but complete data, and this is the only method that will actually iterate through the whole dataset). I can try to clean that API a bit.